runtime/reconcilers/alert.go

```diff
@@ -130,7 +130,7 @@ func (r *AlertReconciler) Reconcile(ctx context.Context, n *runtimev1.ResourceNa
 	adhocTrigger := a.Spec.Trigger
 	specHashTrigger := a.State.SpecHash != specHash
 	refsTrigger := a.State.RefsHash != refsHash && a.Spec.RefreshSchedule != nil && a.Spec.RefreshSchedule.RefUpdate
-	scheduleTrigger := a.State.NextRunOn != nil && !a.State.NextRunOn.AsTime().After(time.Now())
+	scheduleTrigger := a.State.NextRunOn != nil && a.State.NextRunOn.AsTime().Before(time.Now())
```

Comment:
This is not equivalent – specifically, if `NextRunOn == time.Now()`, we should trigger

Reply:
Reverted and fixed linter changes